nomad/job_endpoint_hook_implicit_identities.go

```diff
@@ -31,9 +31,6 @@ func (h jobImplicitIdentitiesHook) Mutate(job *structs.Job) (*structs.Job, []err
 				h.handleConsulService(s, tg)
 				hasIdentity = hasIdentity || s.Identity != nil
 			}
-			if len(t.Templates) > 0 {
-				h.handleConsulTasks(t, tg)
```

Comment:
I think we need to introduce a new implicit identity here when the `consul` block is present at the group/task level.